src/Settings/GtkSettings.vala

```diff
@@ -25,12 +25,19 @@ namespace PantheonTweaks {
         private GLib.KeyFile keyfile;
         private string path;
         private static GtkSettings? instance = null;
+        private bool keyfile_exists = false;
 
         /**
          * GTK should prefer the dark theme or not
          */
         public bool prefer_dark_theme {
-            get { return (get_integer ("gtk-application-prefer-dark-theme") == 1); }
+            get {
+                if (keyfile_exists) {
```

Comment:
Instead of checking for the boolean, let's do the File exists check here as well. You could probably land on a state where the file was not created before, but prefer_dark_theme would always return false until a new settings instance is created :)